src/platform/nrfconnect/OTAImageProcessorImplWiFi.h

```diff
@@ -0,0 +1,63 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "OTAImageProcessorImpl.h"
+#include "wifi/WiFiManager.h"
+
+namespace chip {
+
+namespace DeviceLayer {
+
+class OTAImageProcessorImplWiFi : public OTAImageProcessorImpl
+{
+public:
+    explicit OTAImageProcessorImplWiFi(ExternalFlashManager * flashHandler = nullptr) : OTAImageProcessorImpl(flashHandler) {}
+
+    CHIP_ERROR PrepareDownload() override
+    {
+        CHIP_ERROR err = WiFiManager::Instance().SetLowPowerMode(false);
```

Comment:
The approach is correct though we may think in the future about incorporating events like https://github.com/project-chip/connectedhomeip/commit/7e728bc1bd0275f1dff8d75b2214c745047f6e01 (once that is merged to sdk-connectedhomeip) to do certain actions on OTA state changes to make the solution more extensible.

Reply:
Thanks for pointing this out @Damian-Nordic